crates/xilem_html/src/element/elements.rs

```diff
@@ -0,0 +1,232 @@
+//! Types that wrap [`Element`][super::Element] and represent specific element types.
+//!
+macro_rules! elements {
+    () => {};
+    (($ty_name:ident, $builder_name:ident, $name:literal, $web_sys_ty:ty), $($rest:tt)*) => {
+        element!($ty_name, $builder_name, $name, $web_sys_ty);
+        elements!($($rest)*);
+    };
+}
+
+macro_rules! element {
+    ($ty_name:ident, $builder_name:ident, $name:literal, $web_sys_ty:ty) => {
+        /// A view representing a
+        #[doc = concat!("`", $name, "`")]
+        /// element.
+        pub struct $ty_name<ViewSeq>(crate::Element<$web_sys_ty, ViewSeq>);
+
+        /// Builder function for a
+        #[doc = concat!("`", $name, "`")]
+        /// view.
+        pub fn $builder_name<ViewSeq>(children: ViewSeq) -> $ty_name<ViewSeq> {
+            $ty_name(crate::element($name, children))
```

Comment:
Incidentally, it's possible to use `stringify!($builder_name)` instead of `$name` here, which might save a parameter and reduce duplication, if these names will always be identical.